prover/src/fri/mod.rs

```diff
@@ -0,0 +1,180 @@
+use lambdaworks_math::field::element::FieldElement;
+use lambdaworks_math::field::fields::u64_prime_field::U64PrimeField;
+use lambdaworks_math::polynomial::Polynomial;
+
+const ORDER: u64 = 293;
+pub type F = U64PrimeField<ORDER>;
+pub type FE = FieldElement<F>;
+
+fn fold_polynomial(
+    poly: &Polynomial<FieldElement<F>>,
+    beta: &FieldElement<F>,
+) -> Polynomial<FieldElement<F>> {
+    let coef = poly.coefficients();
+    let even_coef: Vec<FieldElement<F>> = coef
+        .iter()
+        .enumerate()
+        .filter(|(pos, _)| pos % 2 == 0)
+        .map(|(_pos, v)| *v)
```

Comment:
does this do anything?

Reply:
I am filtering the elements that are in even position, so I use enumerate que generate a tupe with (position, value), then I filter by position, and finally I need to go back to the element thrwoing away the position (with the map iterator).